### PR TITLE
fix layout and upgrade ruby version

### PR DESCRIPTION
**Pull request by obie**
## What does this change?

- upgrade Ruby version
- fix basic layout and style issues
- introduce timestamp to post display

## Why are you changing that?

2.4 is very old at this point

## Screenshots

### CODE REVIEW

File: db/structure.sql [hunk 1]
```diff
@@ -2,7 +2,7 @@
 -- PostgreSQL database dump
 --
 
--- Dumped from database version 13.1
+-- Dumped from database version 9.6.20
```
Comment (softr8) [inline db/structure.sql]: I think this file was added by accident, it was generated with an older version of pg and includes some column names that are not referenced anywhere for example: `slack_name`

Comment (fedgut) [reply db/structure.sql]: I regenerated the schema file! Ill add the current version

File: app/controllers/sessions_controller.rb [hunk 2]
```diff
@@ -9,7 +9,7 @@ def create
     developer_info = request.env['omniauth.auth']
 
     developer = Developer.find_or_create_by!(email: developer_info['info']['email']) do |dev|
-      dev.username = developer_info['info']['name']
+      dev.username = developer_info['info']['email'].split('@').first.parameterize
```
Comment (fedgut) [inline app/controllers/sessions_controller.rb:12]: I'm updating this to make sure the greeting doesn't reference the [info][name]

**Review by fedgut — APPROVED** [review]
LGTM! (Added the schema file with proper PG v)

**Pull request merged**